src/app/components/Header/ScriptLink/index.tsx

```diff
@@ -29,25 +44,22 @@ export const getVariantHref = ({
   }
 
   try {
-    return compile(path)(
-      {
-        ...pathParams,
-        variant: `/${variant}`,
-        amp: undefined, // we don't want to link to AMP pages directly
-        nonCanonicalArticleRenderPlatform: undefined, // we don't want to link to AMP (.amp) or APP (.app) for the Optimo article route
-      },
-      {
-        encode: value => value,
-      },
-    );
+    return compile(path, {
+      encode: value => value,
+    })({
+      ...pathParams,
+      variant: `/${variant}`,
+      amp: undefined, // we don't want to link to AMP pages directly
+      nonCanonicalArticleRenderPlatform: undefined, // we don't want to link to AMP (.amp) or APP (.app) for the Optimo article route
+    });
```

Comment:
I find this a wee bit hard to read - could we split it out & make it a bit more readable whilst we're in here?

Something along the lines of: 
```suggestion
    const compilePath = compile(path, {encode: value => value});

    return compilePath({
      ...pathParams,
      variant: `/${variant}`,
      amp: undefined, // we don't want to link to AMP pages directly
      nonCanonicalArticleRenderPlatform: undefined, // we don't want to link to AMP (.amp) or APP (.app) for the Optimo article route
    });
```

Reply:
Yeah makes sense. This thing also confused as the [original implementation](https://github.com/bbc/simorgh/blob/6afbdb5c049ad5af8c89e15102f9f500b745cab0/src/app/legacy/containers/Header/ScriptLink/index.jsx#L32-L42) doesn't seem like it would work? When it was like this and I moved over to .tsx, the compiler was saying it wasn't expecting 2 arguments, so I'm having doubts this worked originally at all